### PR TITLE
Create test.py

### PR DESCRIPTION
**Pull request by PierreGode**

### CODE REVIEW

File: test.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import unittest
```
Comment (PierreGode) [inline test.py:1]: The extra '+' at the beginning of the import statement should be removed.

File: test.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+def add(a, b):
+    return a + b
+
```
Comment (PierreGode) [inline test.py:5]: Consider handling edge cases in the add function, such as adding very large numbers or different data types.

File: test.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+class TestMathOperations(unittest.TestCase):
+    def test_add(self):
+        self.assertEqual(add(2, 3), 5)
+        self.assertEqual(add(-1, 1), 0)
+        self.assertEqual(add(0, 0), 0)
+
+    def test_subtract(self):
+        self.assertEqual(subtract(5, 3), 2)
+        self.assertEqual(subtract(10, 10), 0)
```
Comment (PierreGode) [inline test.py:17]: The test cases could benefit from additional scenarios to ensure robustness, such as testing with negative numbers in the subtract function.

**Comment by PierreGode** [comment]
**AI Pull Request Summary:**
### Pull Request Summary for `gitdemo 2025`

#### Key Changes

- **File Modified**: `test.py`
  
#### Main Purpose of Changes

The primary purpose of this Pull Request is to introduce a set of unit tests for basic arithmetic operations, specifically addition and subtraction. This enhances the testing framework of the repository, ensuring that these operations can be validated for correctness.

#### Specific Functionalities Introduced

The following functionalities were introduced in the `test.py` file:

1. **Arithmetic Functions**:
   - **Addition Function**:
     ```python
     def add(a, b):
         return a + b
     ```
   - **Subtraction Function**:
     ```python
     def subtract(a, b):
         return a - b
     ```

2. **Unit Testing**:
   - A test class `TestMathOperations` is created using `unittest`, which includes two test methods:
     - `test_add`: Tests various cases for the `add` function.
     - `test_subtract`: Tests various cases for the `subtract` function.

The test methods check for expected outcomes using assertions:
- For the `add` function:
  - `add(2, 3)` should return `5`
  - `add(-1, 1)` should return `0`
  - `add(0, 0)` should return `0`
  
- For the `subtract` function:
  - `subtract(5, 3)` should return `2`
  - `subtract(10, 10)` should return `0`
  - `subtract(0, 5)` should return `-5`

3. **Run Tests**:
   - The script includes a conditional statement to run the tests when executed directly:
     ```python
     if __name__ == '__main__':
         unittest.main()
     ```

#### Summary of Changes (Lines Added)

- A total of 21 lines were added to implement the above functionalities. The lines added include the definitions of functions, the test class, and the test cases.

#### Implications and Considerations

- **Testing Coverage**: The addition of unit tests increases the testing coverage for basic mathematical operations, which is essential for maintaining code reliability, especially if further functionalities are added later.
  
- **Integration with Existing Code**: These changes don't modify existing functionalities or existing files but rather add a new testing file. Therefore, there should be no breaking changes or performance impacts on the current application.

- **Dependencies**: The modifications rely on the `unittest` module, which is part of Python’s standard library and does not introduce any new dependencies. 

#### Conclusion

This Pull Request is a positive enhancement to the repository by establishing a foundational testing approach for arithmetic functions, which can be further expanded upon in future contributions. The version containing these corrections and improvements will facilitate better code quality and reliability in the long run.
